services/api/lib/utils.js

```diff
@@ -21,8 +21,37 @@ function formatUser(user) {
   return formatted;
 }
 
+function formatProject(project) {
+  var formatted = {};
+  formatted.id = project.id;
+  formatted.version = project.version;
+  formatted.title = project.title;
+  formatted.remixed_from = project.remixed_from;
+
+  formatted.author = formatUser({
+    id: project.id,
+    username: project.username,
+    created_at: project.user_created_at,
+    updated_at: project.user_updated_at,
+    language: project.user_language,
+    country: project.user_country,
+    staff: project.user_staff,
+    moderator: project.user_moderator
+  });
+
+  formatted.history = {
+    created_at: project.created_at,
+    updated_at: project.updated_at
+  };
+
+  formatted.thumbnail = project.thumbnail;
+
+  return formatted;
+}
+
 exports.register = function(server, options, done) {
   server.method('utils.formatUser', formatUser, { callback: false });
+  server.method('utils.formatProject', formatProject, { callback: false });
```

Comment:
Scaling CPU is pretty simple. If we want to spend time on caching I think I might recommend looking at places where we are I/O bound (network, db, disk) early on.